pkg/logs/client/connection_manager.go

```diff
@@ -149,12 +149,17 @@ func (cm *ConnectionManager) handleServerClose(conn net.Conn) {
 	}
 }
 
-// backoff lets the connection manager sleep a bit
-func (cm *ConnectionManager) backoff(ctx context.Context, retries int) {
-	backoffDuration := backoffUnit * time.Duration(retries)
-	if backoffDuration > backoffMax {
-		backoffDuration = backoffMax
+// backoff implements a randomized exponential backoff in case of connection failure
+// each invocation will trigger a sleep between [2^(retries-1), 2^retries) second
+// the exponent is capped at 9, which translates to max sleep between ~4min and ~8.5min
+func (cm *ConnectionManager) backoff(ctx context.Context, retries uint) {
+	if retries > maxExpBackoffCount {
+		retries = maxExpBackoffCount
 	}
+
+	backoffMax := 1 << retries
+	backoffMin := 1 << (retries - 1)
```

Comment:
nit: I would consider checking `retries > 0` before doing that

Reply:
where this function is called, there is if retries > 0 call this function. but yeah I could add a check